astropy/io/fits/connect.py

```diff
@@ -251,13 +254,20 @@ def read_table_fits(input, hdu=None, astropy_native=False, memmap=False,
             else:
                 t.meta[key] = [value]
 
+
         elif key in t.meta:  # key is duplicate
 
             if isinstance(t.meta[key], list):
                 t.meta[key].append(value)
             else:
                 t.meta[key] = [t.meta[key], value]
 
+        elif is_column_keyword_tdisp(key):
+            # I imagine this format needs to be translated from the TDISP
+            # formatting to the Python string formatting, is there a
+            # function in astropy that does this?
+            t.format = value
```

Comment:
You should do this above, in the columns loop:
```
column = Column(data=data[col.name], name=col.name, copy=False, 
                            format=convert_fits_format_to_python(col.disp))
```
where `convert_fits_format_to_python` does not exist yet ;)